plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -103,11 +162,14 @@ public void setSecrets(Map<String, String> secrets) {
         this.secrets = secrets;
     }
 
-    private String getVariable(String key, Properties prop) {
-        if (prop != null && !prop.isEmpty()) {
-            return prop.getProperty(key, System.getenv(key));
-        } else {
-            return System.getenv(key);
-        }
+    private Optional<String> getVariable(String key, Properties prop) {
+        return prop.containsKey(key) ?
+                Optional.ofNullable(prop.getProperty(key)) : Optional.ofNullable(System.getenv(key));
+    }
+
+    private Optional<String[]> getCommaSeparatedVariables(String key, Properties prop) {
+        Optional<String> setting = getVariable(key, prop);
```

Comment:
optional allows for this oneliner :smile: 
```
return getVariable(key).map(str -> str.split(","));
```

Reply:
Yes that looks way better. Thank you! :)